### PR TITLE
move QgsTolerance::UnitType to Qgis::MapUnitType

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Comment by nyalldawson** [comment]
Nice!

Something about MapUnitType doesn't sit well with me, it sounds a little too associated with the concept of "map units", which are all distance units.

How would you feel about something like "MapToolUnit" instead?

**Comment by 3nids** [comment]
thanks, it sounds good to me, I didn't have any better proposition.
PR updated with `MapToolUnit`

File: src/app/qgsmaptoolsimplify.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ QgsSimplifyUserInputWidget::QgsSimplifyUserInputWidget( QWidget *parent )
 
   // communication with map tool
   connect( mToleranceSpinBox, static_cast < void ( QDoubleSpinBox::* )( double ) > ( &QDoubleSpinBox::valueChanged ), this, &QgsSimplifyUserInputWidget::toleranceChanged );
-  connect( mToleranceUnitsComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, [ = ]( const int index ) {emit toleranceUnitsChanged( ( QgsTolerance::UnitType )index );} );
+  connect( mToleranceUnitsComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, [ = ]( const int index ) {emit toleranceUnitsChanged( ( Qgis::MapToolUnit )index );} );
```
Comment (nyalldawson) [inline src/app/qgsmaptoolsimplify.cpp:59]: Shouldn't this be:
```suggestion
  connect( mToleranceUnitsComboBox, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, [ = ]( int ) {emit toleranceUnitsChanged( mToleranceUnitsComboBox->currentData().value< Qgis::MapToolUnit >();} );
```

File: src/app/qgssnappingwidget.cpp [hunk 2]
```diff
@@ -580,7 +580,7 @@ void QgsSnappingWidget::changeMaxScale( double maxScale )
 
 void QgsSnappingWidget::changeUnit( int idx )
 {
-  QgsTolerance::UnitType unit = static_cast<QgsTolerance::UnitType>( mUnitsComboBox->itemData( idx ).toInt() );
+  Qgis::MapToolUnit unit = static_cast<Qgis::MapToolUnit>( mUnitsComboBox->itemData( idx ).toInt() );
```
Comment (nyalldawson) [inline src/app/qgssnappingwidget.cpp:583]:
```suggestion
  Qgis::MapToolUnit unit = mUnitsComboBox->itemData( idx ).value< Qgis::MapToolUnit >();
```

File: src/core/expression/qgsexpressioncontextutils.h [hunk 3]
```diff
@@ -20,7 +20,9 @@
 #include "qgsfeature.h"
 #include "qgspointlocator.h"
 #include "qgsexpressioncontext.h"
+#include "qgsmapsettings.h"
```
Comment (nyalldawson) [inline src/core/expression/qgsexpressioncontextutils.h:23]: Can't QgsMapSettings be forward declared here? It's only used as a const pointer in this header.

File: src/core/labeling/qgslabelingresults.h [hunk 4]
```diff
@@ -20,6 +20,8 @@
 #include "qgis_sip.h"
 #include "qgslabelposition.h"
 #include "qgscalloutposition.h"
+#include "qgsmapsettings.h"
```
Comment (nyalldawson) [inline src/core/labeling/qgslabelingresults.h:23]: Same here, this could be forward declared

File: src/core/labeling/qgspallabeling.h [hunk 5]
```diff
@@ -33,6 +33,7 @@
 #include "qgsgeometry.h"
 #include "qgsfields.h"
 #include "qgspointxy.h"
+#include "qgsmaplayer.h"
```
Comment (nyalldawson) [inline src/core/labeling/qgspallabeling.h:36]: forward declare

File: src/core/qgis.h [hunk 6]
```diff
@@ -3225,6 +3224,20 @@ class CORE_EXPORT Qgis
     };
     Q_ENUM( SystemOfMeasurement )
 
+    /**
+    * Type of unit of tolerance value from settings.
+    * For map (project) units, use ProjectUnits.
```
Comment (nyalldawson) [inline src/core/qgis.h:3229]:
```suggestion
    * For map (project) units, use MapToolUnit::Project.
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**